src/llmcompressor/modifiers/awq/base.py

```diff
@@ -0,0 +1,678 @@
+import inspect
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Union
+
+import torch
+from compressed_tensors.utils import align_module_device, update_offload_parameter
+from loguru import logger
+from pydantic import ConfigDict
+from torch.nn import Module
+from tqdm import tqdm
+
+from llmcompressor.core import State
+from llmcompressor.modifiers import Modifier
+from llmcompressor.modifiers.utils.pytorch_helpers import run_calibration_forward
+from llmcompressor.pytorch.utils import tensor_forward_with_input_args
+from llmcompressor.utils.fsdp.helpers import get_fsdp_parent
+from llmcompressor.utils.helpers import calibration_forward_context
+from llmcompressor.utils.pytorch.module import (
+    get_layers,
+    get_matching_layer,
+    get_parent_by_name,
+)
+
+__all__ = ["AWQMapping", "AWQModifier"]
+
+
+@dataclass
+class AWQMapping:
+    """
+    Dataclass storing config of activation mappings to smooth
+    The output activations of smooth_layer are input activations
+    into the balance_layers
+
+    `AWQMapping`s are resolved into `ResolvedMapping`s, which
+    retain pointers to the actual `torch.nn.Module`s and additional
+    metadata at runtime
+    """
+
+    smooth_layer: str
+    balance_layers: list[str]
+
+
+DEFAULT_AWQ_MAPPINGS: list[AWQMapping] = [
+    AWQMapping(
+        "re:.*input_layernorm",
+        ["re:.*q_proj", "re:.*k_proj", "re:.*v_proj"],
+    ),
+    # TODO this should only be added if v_proj/o_proj shapes match up, should we check during validation and skip if this is not the case?
```

Comment:
This is the one thing I'm not sure about. The logic for this in AutoAWQ is to only add this mapping if the shapes line up correctly (logic [here](https://github.com/casper-hansen/AutoAWQ/blob/main/awq/models/llama.py#L57-L64)). This is the case for the llama 2 models i've been testing on, but not all of the tiny llama models. Any suggestion on how best to handle for both cases?